### PR TITLE
pep8: fix sanity for 3.12

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/compat/selinux.py
test/units/module_utils/basic/test_argument_spec.py

### CODE REVIEW

File: lib/ansible/module_utils/compat/selinux.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def from_param(cls, strvalue):
         fn.restype = cfg.get('restype', c_int)
 
         # just patch simple directly callable functions directly onto the module
-        if not fn.argtypes or not any(argtype for argtype in fn.argtypes if type(argtype) == base_ptr_type):
+        if not fn.argtypes or not any(argtype for argtype in fn.argtypes if isinstance(argtype, base_ptr_type)):
```
Comment (mattclay) [inline lib/ansible/module_utils/compat/selinux.py]: @nitzmahone Was there a reason for this to use `==` originally instead of `is` or `isinstance`?

Comment (nitzmahone) [reply lib/ansible/module_utils/compat/selinux.py]: Yeah, just changing the `==` to `is` is probably the right thing in this case- `isinstance` is expensive and unnecessary here.

Comment (nitzmahone) [reply lib/ansible/module_utils/compat/selinux.py]: (I don't recall why I had to do all this silliness in the first place, but replacing the `==` with `is` should be semantically equivalent, where `isinstance` could potentially change the behavior in significant ways.

Comment (Akasurde) [reply lib/ansible/module_utils/compat/selinux.py]: Should we ignore `C0123` altogether for this instance?

Comment (mattclay) [reply lib/ansible/module_utils/compat/selinux.py]: No need to ignore it, just replace `==` with `is`, since it's equivalent and passes the test.

**Review by mattclay — APPROVED** [review]

**Pull request merged**